plans/firecracker-backend.md

```
@@ -0,0 +1,331 @@
+- **Linux + KVM only** — acceptable since shed-server runs on Linux
+
+## Prerequisites
+
+After Phase 1 is complete, you have:
+- `internal/backend/backend.go` - Backend interface definition
+- `internal/backend/types.go` - ExecOptions, TerminalSize types
+- `internal/docker/backend.go` - Reference implementation (DockerBackend)
+
+## Architecture
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)